### ISSUE TITLE
Clarify that Retry, VNEG, and Stateless Reset packets don't have packet numbers

### ISSUE THREAD

**Issue by rmarx** [quicwg/qlog#133]
With draft-02, we moved packet_number to PacketHeader and used that field for all types of packets as well. 
However, packet_number is currently a required field, but Retry, Version Negotiation and Stateless Reset packets don't actually use packet numbers.

I don't want to just make the packet_number field generally optional, since imo it shouldn't be for other packet types. Better (best?) solution is to add an explicit comment indicating the field MUST be omitted for the 3 exceptions and MUST be included for all others.

**Comment by LPardue** [comment]
Another option is to define aNumberlessPacketHeader type, which is used in packets that have no numbers

**Comment by marten-seemann** [comment]
The underlying problem here is that some fields only apply to some kind of headers. This is most obvious for the packet number, but it also applies to the Token (which is only carried on Initial packets), the Length (only on long header packets) and the SCID.

If we want to use a strict encoding scheme here, we'll probably have to define a new header for each and every packet type.

**Comment by rmarx** [comment]
At this point in time, I think I'd be fine with just making `packet_number` an optional field and adding comments saying that it of course should be logged if available.

Splitting into several events seems icky just for this (especially not if you'd split it more than once as Marten indicates) and in the end implementers will still choose themselves which fields to log. If they don't include `packet_number` they probably don't need it for their use case (if they do, the lack of it becomes obvious quite immediately).

That doesn't detract libraries/implementations of the qlog schema (e.g., @LPardue's rust crate) to enforce this by including multiple types, but I wouldn't have this in qlog (at least not in my current thinking).

In short: I propose to change to `? packet_number: uint64`.

**Comment by marten-seemann** [comment]
> In short: I propose to change to ? packet_number: uint64.

Sounds reasonable to me.

**Comment by LPardue** [comment]
Optional packet number sounds good too. we already have other optional fields with the conditions of usage stated clearly. This is just another one
